mne/io/neuralynx/neuralynx.py

```diff
@@ -351,13 +305,16 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
         segments_arr[~isgap] = neo_block[0].segments
         segments_arr[isgap] = gap_segments
 
-        # now load data from selected segments/channels via
-        # neo.Segment.AnalogSignal.load() or AnalogSignalGap.load()
+        # now load data for selected segments/channels via
+        # neo.Segment.AnalogSignalProxy.load() or
+        # pad directly as AnalogSignal.magnitude for any gap data
         all_data = np.concatenate(
             [
                 signal.load(channel_indexes=idx).magnitude[
                     samples[0] : samples[-1] + 1, :
                 ]
+                if type(signal) is not AnalogSignal  # then it's AnalogSignalProxy
```

Comment:
```suggestion
                if not isinstance(signal, AnalogSignal)  # then it's AnalogSignalProxy
```
or better, actually check if it's AnalogSignalProxy rather than AnalogSignal?

Reply:
Yeah, was just editing this. Can also check for AnalogSignalProxy. Wasn't sure if it's worth doing an import for that.

Commit coming + sorting the neo import at the write location.